### ISSUE TITLE
dotnet build: False-positive CS8604 in local function

### ISSUE THREAD

**Issue by vladyslav-vasyliev** [dotnet/roslyn#44584]
I've got a false-positive CS8604-warning after VS Build Tools update: `[CS8604] Possible null reference argument for parameter 'timestamp' in 'TimeGuid TimeGuid.MinForTimestamp(Timestamp timestamp)'.`

This one builds without CS8604:
```
[Test]
public void Test()
{
	var now = Timestamp.Now;
	Func();

	TimeGuid Func()
	{
		return TimeGuid.MinForTimestamp(now);
	}
}
```

This one builds without CS8604 too:
```
[Test]
public void Test()
{
	Timestamp now = Timestamp.Now;

	TimeGuid Func()
	{
		return TimeGuid.MinForTimestamp(now);
	}
}
```

This one builds with CS8604 for variable `now` in local function:
```
[Test]
public void Test()
{
	var now = Timestamp.Now;

	TimeGuid Func()
	{
		return TimeGuid.MinForTimestamp(now);
	}
}
```

`Timestamp` is a class.
`Timestamp.Now` is a static property.

Looks like some kind of a bug.

SDK 3.1.300

.csproj conatins these properties:
```
<LangVersion>8</LangVersion>
<Nullable>enable</Nullable>
<IsPackable>false</IsPackable>
<WarningsAsErrors>CS8604;...</WarningsAsErrors>
```

**Comment by RikkiGibson** [comment]
Duplicate of #40157
